### ISSUE TITLE
@material\data-table\component.ts is missing from the TypeScript compilation

### ISSUE THREAD

**Issue by spiotr12** [material-components/material-components-web#4987]
### Steps to reproduce
1. Setup angular application and library
2. Create directive that will import import { MDCDataTable } from '@material/data-table/component'; (see code below)
3. Use that code/library anywhere in angular app
4. Start the app
5. Observe angular console - the error shows up
6. Change anything in the codo to trigger live reload
7. See that app compiles correctly and now it is working in the browser

### Actual behavior
```
ERROR in ./node_modules/@material/data-table/component.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/index.js):
Error: C:\Projects\material-components-angular\node_modules\@material\data-table\component.ts is missing from the TypeScript compilation. Please make sure it i
s in your tsconfig via the 'files' or 'include' property.
The missing file seems to be part of a third party library. TS files in published libraries are often a sign of a badly packaged library. Please open an issue
in the library repository to alert its author and ask them to package the library using the Angular Package Format (https://goo.gl/jB3GVv).
    at AngularCompilerPlugin.getCompiledFile (C:\Projects\material-components-angular\node_modules\@ngtools\webpack\src\angular_compiler_plugin.js:869:23)
    at plugin.done.then (C:\Projects\material-components-angular\node_modules\@ngtools\webpack\src\loader.js:41:31)
    at process._tickCallback (internal/process/next_tick.js:68:7)
ERROR in ./node_modules/@material/data-table/constants.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/index.js):
Error: C:\Projects\material-components-angular\node_modules\@material\data-table\constants.ts is missing from the TypeScript compilation. Please make sure it i
s in your tsconfig via the 'files' or 'include' property.
The missing file seems to be part of a third party library. TS files in published libraries are often a sign of a badly packaged library. Please open an issue
in the library repository to alert its author and ask them to package the library using the Angular Package Format (https://goo.gl/jB3GVv).
    at AngularCompilerPlugin.getCompiledFile (C:\Projects\material-components-angular\node_modules\@ngtools\webpack\src\angular_compiler_plugin.js:869:23)
    at plugin.done.then (C:\Projects\material-components-angular\node_modules\@ngtools\webpack\src\loader.js:41:31)
    at process._tickCallback (internal/process/next_tick.js:68:7)
** Angular Live Development Server is listening on localhost:4220, open your browser on http://localhost:4220/ **
i ｢wdm｣: Failed to compile.
i ｢wdm｣: Compiling...

```

What is interesting the problem is gone after angular app recompile due to change. 

### Expected behavior
No error should appear on first compilation

### Your Environment:

| Software         | Version(s) |
| ---------------- | ---------- |
| MDC Web          | 3.1.0
| Browser          | Chrome
| Operating System | Windows

### Additional context
I'm creating library wrapper for angular 8 that uses directives to wrap Material web components

### Possible solution
Fix tsconfig (?)

**Comment by spiotr12** [comment]
I was able to resolve it by adding
```
  "include": [
    "node_modules/@material/data-table/**/*.ts"
  ]
```
to all my tsconfig.jsonc (lib and app as well)

However this is not ideal in my opinion

**Comment by abhiomkar** [comment]
Can I try importing without referencing the file name?

```js
import {MDCDataTable} from '@material/data-table`;
```

**Comment by spiotr12** [comment]
Hi
What you've proposed did help for the other issue (Data table error while building angular library). However it still fails on first compilation. I've tried both ways: default way to use angular lib as well as setting live reload to the library by changing:
```

    "paths": {
      "my-lib": [
        "projects/my-lib/src"
      ],
      "my-lib/*": [
        "projects/my-lib/src/*"
      ]
    }
```

**Comment by abhiomkar** [comment]
I wouldn't recommend using typescript code directly. `.ts` file shouldn't have been published on NPM repository `@material/data-table`, it's a mistake. PR #4992 should fix it. 

I would recommend using already transpiled JS code.

**Comment by spiotr12** [comment]
Thanks a lot, I'll wait for this and when it is released I will test it if it works

**Comment by abhiomkar** [comment]
Fixed in #4992
